src/controllers/booking.controller.js

```diff
@@ -0,0 +1,56 @@
+import Responses from '../utils/response';
+import bookingService from '../services/booking.service';
+import roomService from '../services/room.service';
+import db from '../models';
+
+/**
+ * Bookings controller
+ */
+class Bookings {
+  /**
+     * Book an accomodation
+     *
+     * @param {Object} req request
+     * @param {Object} res response
+     * @returns {Object} response
+     */
+  async book(req, res) {
+    const {
+      hotelId,
+      arrivalDate,
+      leavingDate,
+      rooms
+    } = req.body;
+
+    const { userId } = res.locals.user;
+
+    const book = {
+      userId,
+      hotelId,
+      arrivalDate,
+      leavingDate
+    };
+    const pResult = rooms.map(async (roomId) => {
```

Comment:
It is hard to know the meaning of `pResult`. It can be renamed to show its intent.

Reply:
I think `promiseResults` can be better

Reply:
It looks to be an array of promises but of which kind?.  promise`Results` leaves us confused.

Reply:
I have changed to `bookedRooms`